src/Compilers/CSharp/Portable/Symbols/SubstitutedNamedTypeSymbol.cs

```diff
@@ -258,6 +267,12 @@ private ArrayBuilder<Symbol> AddOrWrapTupleMembersIfNecessary(ArrayBuilder<Symbo
 
         internal sealed override ImmutableArray<Symbol> GetMembersUnordered()
         {
+            if (!_lazyMembers.IsDefault)
+            {
+                // If we already calculated the ordered members, return it directly.
+                return _lazyMembers.ConditionallyDeOrder();
```

Comment:
I realize the comment for `GetMembersUnordered()` states the order may change from call to call, but this may be the first implementation where that is the case (the order will be different before and after `_lazyMembers` is populated). Is that a concern?

Reply:
> Is that a concern?

We do not expose ```GetMembersUnordered``` to external consumers. However, there is still a chance that an internal implementation somewhere took a dependency on getting items in the same order each time. Reviewing existing call sites might help. What do you think?

Reply:
> Reviewing existing call sites might help.

There are 50 callsites for `NamespaceOrTypeSymbol.GetMembersUnordered()`, and a few of those are methods that return that value so the number of use-sites may be too many to review easily. That said, I skimmed through the direct callers of `GetMembersUnordered()` and didn’t see any obvious issues.

Reply:
@Youssef1313, it looks like the perf issue is allocations from `GetMembers()` and not `GetMembersUnordered()`. Given that, to avoid the risk of breaking callers of `GetMembersUnordered()`, let's revert the change to `GetMembersUnordered()`, and always recalculate that array. Thanks.